libs/velo-external-db-core/src/service/schema.spec.ts

```diff
@@ -1,90 +1,285 @@
+    InputFieldToWixFormatField,
+} from '../utils/schema_utils'
+import { 
+    Table,
+    InputField
+ } from '@wix-velo/velo-external-db-types'
+ 
+const { collectionsListFor } = matchers
 const chance = Chance()
 
 describe('Schema Service', () => {
+    describe('Collection new SPI', () => {
+        test('retrieve all collections from provider', async() => {
+            const collectionCapabilities = {
+                dataOperations: [],
+                fieldTypes: [],
+                collectionOperations: [],
+            }
 
-    test('retrieve all collections from provider', async() => {
-        driver.givenAllSchemaOperations()
-        driver.givenListResult(ctx.dbsWithIdColumn)
+            driver.givenAllSchemaOperations()
+            driver.givenCollectionCapabilities(collectionCapabilities)
+            driver.givenColumnCapabilities()
+            driver.givenListResult(ctx.dbsWithIdColumn)
 
-        await expect( env.schemaService.list() ).resolves.toEqual( schemasListFor(ctx.dbsWithIdColumn, AllSchemaOperations) )
-    })
 
-    test('retrieve short list of all collections from provider', async() => {
-        driver.givenListHeadersResult(ctx.collections)
+            await expect( env.schemaService.list([]) ).resolves.toEqual(collectionsListFor(ctx.dbsWithIdColumn, collectionCapabilities))
+        })
 
+        test('create new collection without fields', async() => {
+            driver.givenAllSchemaOperations()
+            driver.expectCreateOf(ctx.collectionName)
+            schema.expectSchemaRefresh()
 
-        await expect( env.schemaService.listHeaders() ).resolves.toEqual( schemaHeadersListFor(ctx.collections) )
-    })
+            await expect(env.schemaService.create({ id: ctx.collectionName, fields: [] })).resolves.toEqual({
+                collection: { id: ctx.collectionName, fields: [] } 
+            })
+        })
 
-    test('retrieve collections by ids from provider', async() => {
-        driver.givenAllSchemaOperations()
-        schema.givenSchemaFieldsResultFor(ctx.dbsWithIdColumn)
+        test('create new collection with fields', async() => {
+            const fields = [{
+                key: ctx.column.name,
+                type: fieldTypeToWixDataEnum(ctx.column.type),
+            }]
+            driver.givenAllSchemaOperations()
+            schema.expectSchemaRefresh()            
+            driver.expectCreateWithFieldsOf(ctx.collectionName, fields)
+    
+            await expect(env.schemaService.create({ id: ctx.collectionName, fields })).resolves.toEqual({
+                collection: { id: ctx.collectionName, fields }
+            })
+        })
 
-        await expect( env.schemaService.find(ctx.dbsWithIdColumn.map((db: { id: any }) => db.id)) ).resolves.toEqual( schemasListFor(ctx.dbsWithIdColumn, AllSchemaOperations) )
-    })
+        // TODO: move it to schema utils tets
+        test('compareColumnsInDbAndRequest function - add columns', async() => {
```

Comment:
I think you should choose -  move it to `schema.utils.spec` or move this function to be private method in schema service.